multicluster/controllers/multicluster/label_identity_controller.go

```diff
@@ -0,0 +1,347 @@
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: common.DefaultWorkerCount,
+		}).
+		Complete(r)
+}
+
+// namespaceMapFunc handles Namespace update events (Namespace label change) by enqueuing
+// all Pods in the Namespace into the reconciler processing queue.
+func (r *LabelIdentityReconciler) namespaceMapFunc(ns client.Object) []reconcile.Request {
+	podList := &v1.PodList{}
+	r.Client.List(context.TODO(), podList, client.InNamespace(ns.GetName()))
+	requests := make([]reconcile.Request, len(podList.Items))
+	for i, pod := range podList.Items {
+		requests[i] = reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      pod.GetName(),
+				Namespace: pod.GetNamespace(),
+			},
+		}
+	}
+	return requests
+}
+
+// onPodDelete removes the Pod and label identity mapping from the cache, and
+// updates label identity ResourceExport if necessary (the Pod deletion event
+// causes a label identity no longer present in the cluster).
+func (r *LabelIdentityReconciler) onPodDelete(ctx context.Context, podNamespacedName string) error {
+	labelToDelete := r.getLabelToDelete(podNamespacedName)
+	if labelToDelete != "" {
+		if err := r.deleteLabelIdentityResExport(ctx, labelToDelete); err != nil {
+			return err
+		}
+		r.deleteFromLabelCache(podNamespacedName, labelToDelete)
+	}
+	return nil
+}
+
+// onPodCreateOrUpdate updates the Pod and label identity mapping in the cache, and
+// updates label identity ResourceExport if necessary (the Pod creation/update
+// event causes a new label identity to appear in the cluster or a label identity
+// no longer present in the cluster).
+func (r *LabelIdentityReconciler) onPodCreateOrUpdate(ctx context.Context, podNamespacedName, normalizedLabel string) error {
+	labelToAdd, labelToDelete := r.getLabelsToUpdate(podNamespacedName, normalizedLabel)
+	if labelToDelete != "" {
+		if err := r.deleteLabelIdentityResExport(ctx, labelToDelete); err != nil {
+			return err
+		}
+		r.deleteFromLabelCache(podNamespacedName, labelToDelete)
+	}
+	if labelToAdd != "" {
+		if err := r.createLabelIdentityResExport(ctx, labelToAdd); err != nil {
+			return err
+		}
+		r.updateLabelCache(podNamespacedName, labelToAdd)
+	}
+	return nil
+}
+
+// getLabelsToUpdate gets any label identities to be added or to be deleted due to Pod
+// update event. It needs to be protected with labelMutex so that concurrent Pod update
+// events will not interfere with the label cache update calculations.
+func (r *LabelIdentityReconciler) getLabelsToUpdate(podNamespacedName, normalizedLabel string) (labelToAdd string, labelToDelete string) {
+	r.labelMutex.Lock()
+	defer r.labelMutex.Unlock()
+
+	podNames, ok := r.labelToPodsCache[normalizedLabel]
+	if ok {
+		// This is a seen label and there's no other worker trying to delete the ResourceExport
+		// of the label concurrently. Update the cache immediately.
+		podNames.Insert(podNamespacedName)
+		r.podLabelCache[podNamespacedName] = normalizedLabel
+		return "", ""
```

Comment:
We may still need to delete old label of the Pod?

Reply:
You are right. Fixed the issue